src/Analysis/pipeline/parameter_types.py

```diff
@@ -0,0 +1,323 @@
+import abc
+
+class BaseData(metaclass=abc.ABCMeta):
+    """
+    Abbstract class od port types, that define user for
+    validation and translation to string
+    """
+    def __init__(self):
+        self.name = ""
+        """Display name of port"""
+        self.description = ""
+        """Display description of port"""
+    
+    @abc.abstractmethod 
+    def to_string(self, value):
+        """Presentation of type in json yaml"""
+        pass
+
+    def contain(self, data):
```

Comment:
prejmenovat na

	def match_type(self, type_tree)

 ... asi obracena logika nez zde zamysleno
pouziti: 
	 data.match_type(type)
Do komentare to vysvetlit:
Returns True, if 'self' is a data tree or a type tree that is subtree of the type tree 'type'.